### PR TITLE
New pipelines for testing application-connector components

### PR DESCRIPTION
**Pull request by koala7659**
Adding to `kyma-integration` pipelines pre and post submit jobs:  

- `main-kyma-integration-k3d-app-gateway` for testing (Central) Application Gateway
- `main-kyma-integration-k3d-app-conn-validator` for  testing (Central) Application  Connectivity Validator
- `main-kyma-integration-k3d-runtime-agent` for  testing Compass Runtime Agent

- All executed tests are in `tests/components/application-connector`
- Testing platform: `k3d` installed in GCP VM
- Minimal Kyma installation for both OS and SKR setup

- The `run if changed ` rule fires pipelines after files are changed at locations: 
    - `resources/application-connector` fires `pre-main-kyma-integration-k3d-app-gateway` and `pre-main-kyma-integration-k3d-app-conn-validator`
    - `resources/compass-runtime-agent` fires `pre-main-kyma-integration-k3d-runtime-agent`

### CODE REVIEW

File: templates/data/generic_component_data.yaml [hunk 1]
```diff
@@ -25,105 +25,6 @@ templates:
...
-                    - "jobConfig_generic_component_kyma"
-                    - "extra_refs_test-infra"
-                  preConfigs:
-                    global:
-                      - "jobConfig_presubmit"
-                  postConfigs:
-                    global:
-                      - "jobConfig_postsubmit"
-                      - "disable_testgrid"
-      - to: ../../prow/jobs/kyma/components/uaa-activator/uaa-activator-generic.yaml
```
Comment (koala7659) [inline templates/data/generic_component_data.yaml]: Why did you remove this job? Isn't a mistake?

**Review by majakurcius — DISMISSED** [review]

File: prow/scripts/cluster-integration/kyma-integration-k3d-app-connector-components-os.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+---
```
Comment (franpog859) [inline prow/scripts/cluster-integration/kyma-integration-k3d-app-connector-components-os.yaml:1]: It looks like these two yamls, I mean this one and the `...skr.yaml` one are duplicates of [these two files](https://github.com/kyma-project/kyma/tree/main/tests/components/application-connector/resources/installation-config). If it's possible I'd suggest using only one pair of these files to avoid desynchronization problems

**Review by franpog859 — APPROVED** [review]

**Review by franpog859 — APPROVED** [review]

**Review by koala7659 — COMMENTED** [review]
LGTM

**Review by Ressetkk — DISMISSED** [review]

**Review by klaudiagrz — APPROVED** [review]

**Review by klaudiagrz — APPROVED** [review]

**Review by Ressetkk — APPROVED** [review]
